code/modules/mob/living/simple_animal/hostile/hostile.dm

```diff
@@ -83,26 +83,21 @@
 
 //////////////HOSTILE MOB TARGETTING AND AGGRESSION////////////
 
-
 /mob/living/simple_animal/hostile/proc/ListTargets()//Step 1, find out what we can see
 	. = list()
 	if(!search_objects)
 		var/list/Mobs = hearers(vision_range, targets_from) - src //Remove self, so we don't suicide
 		. += Mobs
 
-		var/hostile_machines = list()
-		for(var/mecha in mechas_list)
-			hostile_machines += mecha
-		for(var/obj/machinery/porta_turret/P in machines)
-			hostile_machines += P
-		for(var/M in hostile_machines)
-			if(get_dist(M, targets_from) <= vision_range && can_see(targets_from, M, vision_range))
-				. += M
+		var/static/hostile_machines = typecacheof(/obj/machinery/porta_turret, /obj/mecha)
```

Comment:
<i>nag</i> this should be ```var/static/list/hostile_machines``` :P 
(or we could do what /vg/ did with their ```string``` ```int``` etc. defines and have ```#define typecache list```)